pkg/util/reflector/prometheus/prometheus.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheus
+
+import (
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Package prometheus sets the cache DefaultMetricsFactory to produce
+// prometheus metrics. To use this package, you just have to import it.
+
+func init() {
+	cache.SetReflectorMetricsProvider(prometheusMetricsProvider{})
+}
+
+type prometheusMetricsProvider struct{}
+
+func (_ prometheusMetricsProvider) NewListsMetric(name string) cache.CounterMetric {
+	lists := prometheus.NewCounter(prometheus.CounterOpts{
+		Subsystem: name,
```

Comment:
> You might want to consider making this a label, so that we can do aggregations or bulk queries in the dashboard (e.g. see lists done by all reflectors) more easily

Maybe later.  I'm not yet convinced that all reflectors are the same (go into the same larger aggregation bucket"